llvm/lib/IR/Verifier.cpp

```diff
@@ -4392,6 +4392,12 @@ void Verifier::visitAllocaInst(AllocaInst &AI) {
     verifySwiftErrorValue(&AI);
   }
 
+  if (TT.isAMDGPU()) {
+    Check(AI.getAddressSpace() == DL.getAllocaAddrSpace() ||
+              AI.getAddressSpace() == 0,
```

Comment:
```suggestion
    AS = I.getAddressSpace();
    Check(AS == AMDGPUAS::PRIVATE_ADDRESS || AS == AMDGPUAS::FLAT_ADDRESS,
```

Maybe shouldn't trust the datalayout so much

Reply:
We can't include `AMDGPU.h` where `AMDGPUAS` is defined here.

Reply:
Oh we already have the header included.